### ISSUE TITLE
ab-decoder with Quorum

### ISSUE THREAD

**Issue by paliwalg** [ConsenSys/abi-decoder#2]
Does it works with Quorum. Since Quorum is built on top of ethereum it should work.
I tried and none of these two methods decodeMethod & decodeLogs works.

Thanks

**Comment by Moejoe90** [comment]
I'm not really part of the project I'm considering to use this library for decoding some transactions, however just to help you out or set you to the right path. 
This library, in theory, decodes ethereum transactions based on smart contract ABI and data input from transactions.
I'm not sure what you mean exactly with your question, if you mean analyzing Quorum smart contracts then it should work. However, if you mean decode Quorum transactions (I have no background Quorum) then that means they have to follow the same rules as ethereum in decoding and I highly doubt that.

**Comment by paliwalg** [comment]
No worries, was actually the issue with Quorum. In case of private transactions , the payload contains only the hash that’s why decoding is not going to work there. Thanks

**Comment by Moejoe90** [comment]
Yes, that is indeed the case, internal transactions are really an interesting game, but to find them you will most likely need to instrument the EVM and decode the whole transaction again which in any case will have more information than you need to decode internal transactions or know what they are doing :)
